### PR TITLE
Admin should be able to assign driver from dropdown

### LINKED ISSUE #6: FEATURE: Admin should be able to select driver from dropdown for assigning ride request
### As a...

admin

### I can...

assign drivers to ride requests by selecting from a dropdown

### So that...

I dont have to look up the driver numbers and can select from a list

### Discussion

Right now admins are able to assign drivers to rider's ride requests, but they currently use the ID to shifts.

image

What would be better is a dropdown menu that contains a list of available shifts, with both id and driver name:

```
1 - Andrew Peng
2 - Phill Conrad
```

What would be even better but not a requirement for this feature is either:

* The dropdown shows more information about the shift

```
1 - Andrew Peng - Monday 11:00AM-12:00PM
2 - Phill Conrad - Thursday 01:00PM-3:00PM
```

* OR we only show the shifts that match the day and/or time of the ride request

### Acceptance Criteria

- [x] When assigning a driver/shift to a ride request, an admin can select from a list of shifts in a dropdown menu as opposed to a text field. This dropdown menu can match any of the solutions presented in the discussion.

### Implementation Todos

Frontend:
- [x]  Edit `RideAssignDriverForm.js` and instead of a text field for `Shift Id`, there is a dropdown.

Backend:
- [x] If you decide you want to filter all of the shifts by their day/time you can add a new method to `ShiftController.java`

### PR DESCRIPTION
**Pull request by Ononymous**
# As a...

- admin

# I can...

- assign drivers to ride requests by selecting from a dropdown

# So that...

- I dont have to look up the driver numbers and can select from a list

image

Changed to dropdown, used useBackend to get all drivers and shifts, and mapped the driver names to each shift using driver id.

image

Closes #6

### CODE REVIEW

File: frontend/src/main/components/Ride/RideAssignDriverForm.js [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+                    id: shift.id, 
+                    driverName: driverMap.get(shift.driverID),
+                    day: shift.day,
+                    shiftStart: shift.shiftStart,
+                    shiftEnd: shift.shiftEnd,
+                }]);
+            });
+        }
+    }, [drivers, shifts]);
+    // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/components/Ride/RideAssignDriverForm.js]:
```suggestion
    // Stryker restore all
```

It turns out `enable all` isn't a thing.

File: frontend/src/main/components/Ride/RideAssignDriverForm.js [hunk 2]
```diff
@@ -17,7 +64,7 @@ function RideAssignDriverForm({ initialContents, submitAction, buttonLabel = "As
         { defaultValues: initialContents }
     );
     // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/components/Ride/RideAssignDriverForm.js]:
```suggestion
    // Stryker restore all
```

**Review by pconrad — CHANGES_REQUESTED** [review]
LGTM overall...  one thing to fix (in two places)

**Review by Tommygithubaccount123 — DISMISSED** [review]
LGTM

**Comment by Ononymous** [comment]
> LGTM overall... one thing to fix (in two places)

Fixed stryker enable

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**